### DISCUSSION TITLE
Convert 2sxc app to Oqtane's SPA

### DISCUSSION THREAD

**Discussion by enfJoao** [2sic/2sxc#3516] [Q&A]
Hey.

Does 2sxc plan to support Oqtane's SPA framework and interactive mode to discard javascript for C#?

Or does it already?

**Comment by iJungleboy** [comment]
I assume you're talking about the Edit UIs etc.

This is certainly not planned ;)

The C# Blazor model has a LOT of drawbacks - and changing it would make 2sxc incompatible with DNN.

IMHO Blazor is the next evolution of Razor, in terms of the component-model.

But many other things it introduced - such as running .net in the browser - are IMHO not a good idea. 
A lot of things must happen for this to be possible, and these things have a high cost in terms of Bandwidth and more.

The .net-in-browser appealed to C# developers who were afraid to learn JS SPAs.
But as we can see in every new release, Microsoft is aware of the problem and is making Blazor much more JS friendly than it used to be.

The new "Static Rendering" which was introduced in .net 8 is a clear indication that Microsoft also found it's way "back" :)

**Comment by enfJoao** [comment]
Not just about the edit UIs, but app development itself. Using both DNN and Oqtane, there are a couple issues preventing app creation for both CMSs:
1. Libraries exclusive to either .net framework or .net core (cookie management for example in system.web)
2. Javascript that cannot be placed in external libraries (contains razor code, for example). Blazor does not isolate pages or apps, so they all get added to the DOM/whatever and switching between pages causes wild issues. It's almost like a single Oqtane install is meant for a single app with a very limited purpose.
Blazor proposes the js removal in favor of C#, but I wonder how would this work in a 2sxc app (not an oqtane module).

**Comment by iJungleboy** [comment]
So the road forward IMHO is clearly Blazor on the server.

The component model allows building things which are quite nice.
But in the end we would still generate HTML etc.

Once Server-Side is really "cemented" we can look into providing Blazor as an optional renderer (in addition to Razor).
But I think this will take another 1-2 years.

**Comment by enfJoao** [comment]
I'll pick this up to avoid creating another unnecessary discussion topic.
I've not been able to solve this for a long time, and perhaps it's just a perspective problem.
From what I could gather, Blazor's SPA loads everything into the DOM once. On the other hand, 2sxc does not support blazor on the client, so all @code {} stuff is off limits for now.
So this leaves this question:
https://stackoverflow.com/questions/79302830/detect-page-change-with-2sxc-app-in-blazor-oqtane

How do I run a script when a page loads/shows. Not the first time it loads, but every time I return to it. Oqtane only respects the script "on page load" the first time it renders it.

Sorry for insisting on this question @iJungleboy , but it's a hacky implementation I use on a lot of apps. For example, to read a cookie, fill a text input and filter a table on page load (not on server, but on client). On DNN this happens by default, but on Oqtane I failed miserably.
